cmd/spire-server/cli/agent/list.go

```diff
@@ -0,0 +1,99 @@
+package agent
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	"github.com/spiffe/spire/proto/api/registration"
+	"github.com/spiffe/spire/proto/common"
+
+	"golang.org/x/net/context"
+)
+
+//ListConfig holds configuration for ListCLI
+type ListConfig struct {
+	// Socket path of registration API
+	RegistrationUDSPath string
+}
+
+// Validate will perform a basic validation on config fields
+func (c *ListConfig) Validate() (err error) {
+	if c.RegistrationUDSPath == "" {
+		return errors.New("a socket path for registration api is required")
+	}
+	return nil
+}
+
+//ListCLI command for listing attested nodes
+type ListCLI struct {
+	RegistrationClient registration.RegistrationClient
+	NodeList           []*common.AttestedNode
+}
+
+func (ListCLI) Synopsis() string {
+	return "List agent spiffeIDs"
```

Comment:
nit: maybe a little more friendly as `List attested agents and their SPIFFE IDs`